docs/destinations/forwarding-events/event-forwarding-monitoring-and-troubleshooting/index.md

```diff
@@ -250,3 +250,163 @@ FROM event_forwarding_failures
 WHERE "$path" > 's3://{BUCKET_NAME}/{PIPELINE_NAME}/partitioned/com.snowplowanalytics.snowplow.badrows.event_forwarding_error/2025-07-29-16'
 AND "$path" < 's3://{BUCKET_NAME}/{PIPELINE_NAME}/partitioned/com.snowplowanalytics.snowplow.badrows.event_forwarding_error/2025-07-29-20'
 ```
+
+### Querying failed events on GCP BigQuery
+
+For GCP deployments, you can use BigQuery to query your failed events using external tables that reference the cloud storage files.
+
+**1. Create a dataset**
+
+First, create a dataset to organize your failed event tables:
+
+```sql
+CREATE SCHEMA snowplow_failed_events
+OPTIONS (
+  description = "Dataset for Snowplow failed event analysis",
+  location = "EU"  -- Should match the location of your bad rows bucket
+);
+```
+
+**2. Create an external table and load the data**
+
+To make the logs easier to query, run the following query to create an external table:
+
+```sql
+  CREATE OR REPLACE EXTERNAL TABLE snowplow_failed_events.event_forwarding_failures (
+    schema STRING,
+    data STRUCT<
+      failure STRUCT<
+        errorCode STRING,
+        errorMessage STRING,
+        errorType STRING,
+        latestState STRING,
+        timestamp TIMESTAMP
+      >,
+      payload STRING,
+      processor STRUCT<
+        artifact STRING,
+        version STRING
+      >
+    >
+  )
+  OPTIONS (
+    description="Event Forwarding failures",
+    format="NEWLINE_DELIMITED_JSON",
+    ignore_unknown_values=true,
+    uris=["gs://{BUCKET}/partitioned/com.snowplowanalytics.snowplow.badrows.event_forwarding_error/*"]
+  );
+```
+
+**3. Explore failure records**
+
+Use the query below to view a sample of failure records:
+
+```sql
+SELECT
+    data.failure.timestamp,
+    data.failure.errorType,
+    data.failure.errorCode,
+    data.failure.errorMessage,
+    data.processor.artifact,
+    data.processor.version,
+    data.failure.latestState, -- these are last because they can be quite large
```

Comment:
I copied this comment from Athena examples, it says `latestState` is last...but is it true as we still query payload at the end?

Reply:
OK, I'm dumb 🤦 Makes sense!